x-pack/plugins/observability_solution/observability/public/pages/alert_details/components/header_actions.tsx

```diff
@@ -78,7 +84,7 @@ export function HeaderActions({ alert, alertStatus, onUntrackAlert }: HeaderActi
       ? [
           {
             alertId: alert?.fields[ALERT_UUID] || '',
-            index: '.internal.alerts-observability.metrics.alerts-*',
+            index: alertIndex || '',
```

Comment:
@benakansara, in what use case the `alertIndex` could be unavailable?

Reply:
@fkanout The `alertDetail` can be undefined [here](https://github.com/benakansara/kibana/blob/85d822834467c3c674f18d13819f66b1f15bd14e/x-pack/plugins/observability_solution/observability/public/pages/alert_details/alert_details.tsx#L243) which can lead to `alertIndex` being undefined. We can perhaps improve it by checking `alertDetail` first, and hiding header actions menu if it is undefined.